kernel_gateway/gatewayapp.py

```diff
@@ -213,17 +258,47 @@ def init_webapp(self):
         '''
         # Redefine handlers off the base_url path
         handlers = []
-        for handler in (
-            default_kernel_handlers + 
-            default_kernelspec_handlers +
-            default_base_handlers
-        ):
-            # Create a new handler pattern rooted at the base_url
-            pattern = url_path_join(self.base_url, handler[0])
-            # Some handlers take args, so retain those in addition to the
-            # handler class ref
-            new_handler = tuple([pattern] + list(handler[1:]))
-            handlers.append(new_handler)
+        if self.api:
```

Comment:
We're definitely going to do some refactoring--the kernel spec resides in the notebook, but we're currently set to only read that after the KernelManager is already created.